addons/l10n_dk_nemhandel/models/account_edi_xml_oioubl_21.py

```diff
@@ -0,0 +1,250 @@
+    'S': 'StandardRated',
+    'Z': 'ZeroRated',
+    'G': 'ZeroRated',
+    'O': 'ZeroRated',
+    'K': 'ReverseCharge',
+    'L': 'ZeroRated',
+    'M': 'ZeroRated',
+}
+
+
+def format_vat_number(partner):
+    vat = (partner.vat or '').replace(' ', '')
+    if vat[:2].isnumeric():
+        vat = partner.country_code.upper() + vat
+    return vat
+
+
+class AccountEdiXmlOIOUBL21(models.AbstractModel):
+    _name = "account.edi.xml.oioubl_21"
+    _inherit = 'account.edi.xml.ubl_21'
+    _description = "OIOUBL 2.1"
+
+    # Data validation Schematron available at the following URL:
+    # https://rep.erst.dk/git/openebusiness/common/-/tree/master/resources/Schematrons/OIOUBL
+
+    # -------------------------------------------------------------------------
+    # EXPORT
+    # -------------------------------------------------------------------------
+
+    def _export_invoice_filename(self, invoice):
+        return f"{invoice.name.replace('/', '_')}_oioubl_21.xml"
+
+    def _export_invoice_vals(self, invoice):
+        # EXTENDS account.edi.xml.ubl_21
+        vals = super()._export_invoice_vals(invoice)
+        vals['vals'].update({
+            'customization_id': 'OIOUBL-2.1',
+            # ProfileID is the property that define which documents the company can send and receive
+            # 'Procurement-BilSim-1.0' is the simplest one: invoice and bill
+            # https://www.oioubl.info/documents/en/en/Guidelines/OIOUBL_GUIDE_PROFILES.pdf
+            'profile_id': 'Procurement-BilSim-1.0',
+            'profile_id_attrs': {
+                'schemeID': 'urn:oioubl:id:profileid-1.6',
+                'schemeAgencyID': DANISH_NATIONAL_IT_AND_TELECOM_AGENCY_ID,
+            }
+        })
+        vals['vals'].setdefault('document_type_code_attrs', {}).update({
+            'listID': 'urn:oioubl:codelist:invoicetypecode-1.2',
+            'listAgencyID': DANISH_NATIONAL_IT_AND_TELECOM_AGENCY_ID,
+        })
+
+        return vals
+
+    def _get_partner_party_vals(self, partner, role):
+        # EXTENDS account.edi.xml.ubl_20
+        vals = super()._get_partner_party_vals(partner, role)
+
+        if partner.nemhandel_identifier_type and partner.nemhandel_identifier_value:
+            prefix = 'DK%s' if partner.nemhandel_identifier_type == '0184' else '%s'
+            vals.update({
+                'endpoint_id': prefix % partner.nemhandel_identifier_value,
+                'endpoint_id_attrs': {'schemeID': partner.nemhandel_identifier_type},
+            })
+
+        for party_tax_scheme in vals['party_tax_scheme_vals']:
+            # the doc says it could be empty but the schematron says otherwise
+            # https://www.oioubl.info/Classes/en/TaxScheme.html
+            party_tax_scheme.update({
+                'tax_scheme_vals': {
+                    'id': 63,
+                    'id_attrs': {'schemeID': 'urn:oioubl:id:taxschemeid-1.5'},
+                    'name': 'Moms',
+                },
+            })
```

Comment:
This part would more properly belong to the `_get_partner_party_tax_scheme_vals_list` method below :smile: 
But I suggest you refactor the code to use the new helpers instead, so this would all be part of `_get_party_node`.